ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Storage/ChainDB/Impl/ChainSel.hs

```diff
@@ -837,14 +844,30 @@ chainSelectionForBlock cdb@CDB{..} blockCache hdr punish = electric $ do
           $ getSuffix
           $ getChainDiff vChainDiff
         (curChain, newChain, events, prevTentativeHeader) <- atomically $ do
-          curChain  <- readTVar         cdbChain -- Not Query.getCurrentChain!
+          InternalChain curChain curChainWithTime <- readTVar cdbChain
+             -- Not Query.getCurrentChain!
           curLedger <- LgrDB.getCurrent cdbLgrDB
           case Diff.apply curChain chainDiff of
             -- Impossible, as described in the docstring
             Nothing       ->
               error "chainDiff doesn't fit onto current chain"
             Just newChain -> do
-              writeTVar cdbChain newChain
+              let lcfg             = configLedger cdbTopLevelConfig
+                  diffWithTime     =
+                    -- the new ledger state can translate the slots of the new
+                    -- headers
+                    Diff.map
+                      (mkHeaderWithTime
```

Comment:
See https://github.com/IntersectMBO/ouroboros-consensus/issues/1301#issuecomment-2573367631 for a discussion of this "undesirable" calculation of a time